RUFAS/graph_generator.py

```diff
@@ -150,10 +151,84 @@ def generate_graph(
         except Exception as e:
             raise e
 
+    def prepare_plot_data(self, filtered_pool: Dict[str, Dict[str, List[Any]]],
+                          graph_details: Dict[str, str | List[str]],
+                          ) -> Tuple[Dict[str, List[int | float]], List[Dict[str, str | Dict[str, str]]]]:
+        """Extracts the values from the filtered_pool data and converts them a dictionary
+        that graph_generator can more readily handle and records logs, warnings, and errors for
+        Output Manager to log.
+
+        Parameters
+        ----------
+        filtered_pool : Dict[str, pool_element_type]
+            The filtered pool of variables that the user wants to graph.
+        graph_details: Dict[str, str]
+            A dictionary containing details/metadata about the graph.
+
+        Returns
+        -------
+        Tuple[Dict[str, List[int | float]], List[Dict[str, str | Dict[str, str]]]]
+            A tuple containing the formatted data that can more readily be plotted by
+            graph_generator and the logs, warnings, and errors to be reported to OutputManager.
+        """
+        info_map = {
+            "class": self.__class__.__name__,
+            "function": self.prepare_plot_data.__name__,
+        }
+        selected_variables = graph_details.get("variables")
+        title = graph_details.get("title")
+        log_pool = []
+        prepared_pool = {}
+        for index, key in enumerate(filtered_pool.keys()):
+            values: List[Any] = filtered_pool[key]["values"]
+            is_data_in_dict = isinstance(values[0], dict)
+            if is_data_in_dict:
+                if not selected_variables:
+                    error = {"error": f"Can't plot {title} data set", "message": f"No selected variables for {key}.",
+                             "info_map": info_map}
+                    log_pool.append(error)
```

Comment:
Simplify, we add documentary variables when the statement is hard to understand. `error = error` doesn't make the code more readable.
```suggestion
                    log_pool.append({"error": f"Can't plot {title} data set", "message": f"No selected variables for {key}.",
                             "info_map": info_map})
```

Reply:
Makes sense. Updated.